rs/messaging/src/routing/stream_handler.rs

```diff
@@ -934,24 +936,34 @@ impl StreamHandlerImpl {
                 Accept(msg_cycles)
             }
 
-            // Receiver is not and was not (according to `migrating_canisters`) recently
-            // hosted by this subnet.
             host_subnet => {
-                error!(
-                    self.log,
-                    "{}: Dropping misrouted message (receiver {} is hosted by {:?}): {:?}",
-                    CRITICAL_ERROR_RECEIVER_SUBNET_MISMATCH,
-                    msg.receiver(),
-                    host_subnet,
-                    msg
-                );
-                self.observe_inducted_message_status(
-                    msg_type,
-                    LABEL_VALUE_RECEIVER_SUBNET_MISMATCH,
-                );
-                self.metrics.critical_error_receiver_subnet_mismatch.inc();
-                // Cycles are lost.
-                Accept(msg.cycles())
+                if matches!(msg, RequestOrResponse::Request(_)) {
```

Comment:
Nit: I would have kept the previous structure, as it made it clearer that we have two branches at the same level as the two branches above.

```
// Request receiver not hosted by this subnet...
_ if matches!(msg, RequestOrResponse::Request(_)) => {
    ...
}

// Responses to originator not hosted by this subnet is dropped.
host_subnet => {
    ...
}
```